n3fit/src/n3fit/layers/rotations.py

```diff
@@ -91,6 +91,28 @@ def call(self, pdf_raw):
         # Concatenating destroys the batch index so we have to regenerate it
         return op.batchit(ret)
 
+class AddPhoton(MetaLayer):
+    """
+    Changes the value of the photon to non-zero.
+
+    Both input and output have shape (1, None, 14). However,
+    the input has a photon which is set to zero, while for the
+    output it is set to its real value.
+    """
+
+    def __init__(self, output_dim=14, **kwargs):
+        self.output_dim = output_dim
+        super().__init__(**kwargs)
+    
+    def call(self, pdfs, pdf_ph):
+        x = op.transpose(pdfs)
+        pdf_ph_t = op.transpose(pdf_ph)
+        pdf_raw_list = [x[i] for i in range(14)]
+        pdf_raw_list[0] = pdf_ph_t[0]
+        ret = op.concatenate(x)
```

Comment:
It would be better to pass `pdf_ph` with a shape compatible with `pdfs` (whatever it is, I don't know right now, probably `(1, 13, x)` or something) so that you can reduce this operation to `tf.stack[pdf_ph, pdfs])`

Reply:
This you can't do, since you need to slice the photon out of `pdfs` first.

But I agree on something like `tf.stack[pdf_ph, pdfs[1:]])`

Reply:
However, notice that my proposal was not to hard-code any number for flavor, that was the meaning of removing the array of assignments (plus dropping a bunch of code that could be shortened).

So not:
```py
pdf_raw_list = [x[i] for i in range(14)]
```
but:
```py
pdf_raw_list = list(x)
```
The difference is minimal, but the important part is that you don't rely on the God given number `14` (undocumented values are always an issue per se).